lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -72,50 +72,95 @@ void SetMessageManagerHandler::SetAmfSessionContext(
-        std::string dnn =
-            request_cpy.common_context().apn();  // may not required for demo-1
-        std::string session_id = id_gen_.gen_session_id(imsi);
-
-        MLOG(MDEBUG) << "Requested session from UE with IMSI: " << imsi
-                     << " Generated sessioncontext ID" << session_id;
-        /*reach complete message from proto message*/
-        SessionConfig cfg = m5g_build_session_config(request_cpy);
-
-        /*check if it's initial message*/
-        if ((cfg.rat_specific_context.m5gsm_session_context().rquest_type() ==
-             INITIAL_REQUEST) &&
-            (cfg.common_context.sm_session_state() == CREATING_0)) {
-          /* it's new UE establisment request and need to create the session
-           * context
-           */
-          MLOG(MDEBUG)
-              << "AMF request type INITIAL_REQUEST and session state CREATING";
-          /* Read the SessionMap from global session_store,
-           * if it is not found, it will be added w.r.t imsi
-           */
+
+  /* Read the proto message and check for state. Get the config out of proto.
+   * if state is CREATING - New session to be created
+   * if state is RELEASE  - The session to be delered
+   */
+  // Fetch complete message from proto message
+  SessionConfig cfg = m5g_build_session_config(request_cpy);
+  // extract values from proto
+  auto imsi       = request_cpy.common_context().sid().id();
+  std::string dnn = request_cpy.common_context().apn();
+  if (cfg.common_context.sm_session_state() == RELEASED_4) {
+    if (cfg.common_context.sm_session_version() != 0) {
+      /* This is a serious sync error, as AMF passing wrong version values
+       * for this session to release. As session version can never be 0.
+       * Print the error message and keep releasing the respective session
+       */
+      MLOG(MERROR) << "Wrong version received from AMF for IMSI " << imsi
+                   << " but continuing release request";
+    }
+    MLOG(MINFO) << "Release request for session from IMSI: " << imsi << " DNN "
+                << dnn;
+    // Requested message from AMF to release the session
+    m5g_enforcer_->get_event_base().runInEventBaseThread(
+        [this, cfg, imsi, dnn]() {
           auto session_map = session_store_.read_sessions({imsi});
-          send_create_session(session_map, imsi, session_id, cfg);
-        }
-      });
+          initiate_release_session(session_map, dnn, imsi);
+        });
+  }
+  // Else the requested message is for establishment or modification
+  else {
+    // The Event Based main_thread invocation and runs to handle session state
+    m5g_enforcer_->get_event_base().runInEventBaseThread([this, cfg, imsi,
+                                                          dnn]() {
+      std::string session_id = id_gen_.gen_session_id(imsi);
+      MLOG(MINFO) << "Requested session from UE with IMSI: " << imsi
+                  << " Generated session context ID " << session_id;

```

Comment:
```      
	MLOG(MINFO) << " Generated session " << session_id;
```

Reply:
Shud't we display the log,  log is to be removed ?

Reply:
Ok got it corrected the log.